src/engraving/libmscore/staff.cpp

```diff
@@ -790,6 +790,33 @@ Fraction Staff::currentKeyTick(const Fraction& tick) const
     return Fraction::fromTicks(_keys.currentKeyTick(tick.ticks()));
 }
 
+//---------------------------------------------------------
+//   Staff::addMissingInitKeyForTransposingInstrument
+//
+//    if there is no key signature at the begining
+//    of the score, add written C major
+//---------------------------------------------------------
+
+void Staff::addMissingInitKeyForTransposingInstrument()
```

Comment:
A more appropriate place for this function is CompatUtils - there it should be declared as a static function taking Staff as an argument:

CompatUtils::addMissingInitKeyForTransposingInstrument(Score* score, Staff* staff)

Reply:
I moved it to CompatUtils, it is now there, called only from `doCompatibilityConversions` as `CompatUtils::addMissingInitKeyForTransposingInstrument(MasterScore* score)`

I also realized, that 4.1 files can be affected too, if user opened old file in 4.1 and saved it.

So all files before 4.2 are checked.

I reverted first attempt. If this one is OK, I can drop it from PR, if needed.

Reply:
> 
@mercuree, I just tested atonal at first measure, 4.1.1 and master.
Cannot reproduce behaviour, You mentioned.

Reply:
@sammik 
Checked twice. Musescore 4.1.1
1. Create from template -> Treble clef -> Done
2. Save score
3. Set open/atonal key at first measure
4.  Delete first measure
5. Save score

Now if you open mscx (inside mscz archive), you will not find `<KeySig>` element inside.
However it will not crash if you try to change instrument, because it crashes with transposing instrument (do the same with soprano sax, save and reopen file and it will crash on change instrument)

Reply:
@sammik Thank You! New issue is here #19063